internal/git/config.go

```diff
@@ -0,0 +1,145 @@
+package git
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/charmbracelet/log"
+)
+
+// Config provides access to Git configuration in the current context.
+type Config struct {
+	log  *log.Logger
+	dir  string
+	env  []string
+	exec execer
+}
+
+// ConfigOptions configures the behavior of a [Config].
+type ConfigOptions struct {
+	// Dir specifies the directory to run Git commands in.
+	// Defaults to the current working directory if empty.
+	Dir string
+
+	// Env specifies additional environment variables
+	// to set when running Git commands.
+	Env []string
+
+	// Log used for logging messages to the user.
+	// If nil, no messages are logged.
+	Log *log.Logger
+
+	exec execer
+}
+
+// NewConfig builds a new [Config] object for accessing Git configuration.
+func NewConfig(opts ConfigOptions) *Config {
+	exec := opts.exec
+	if exec == nil {
+		exec = _realExec
+	}
+
+	if opts.Log == nil {
+		opts.Log = log.New(io.Discard)
+	}
+
+	return &Config{
+		log:  opts.Log,
+		dir:  opts.Dir,
+		env:  opts.Env,
+		exec: exec,
+	}
+}
```

Comment:
I would expect an Option pattern here and not this

More information here

https://dev.to/kittipat1413/understanding-the-options-pattern-in-go-390c

Your code is fine, but the option pattern will make everything simpler

Reply:
Thanks for the tip, @ccoVeille. I don't think the boilerplate/cost of functional options is worth it here given that the list of callers is tightly controlled—seeing as it's an internal package. The API needs to at least be public to justify the cost of functional options. (FWIW, other entry points in this package also use Options structs instead of functional options.)